core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -18,8 +18,15 @@
 syntax = "proto3";
 package org.apache.spark.status.protobuf;
 
+/**
+ * Developer guides:
+ *   - Coding style: https://developers.google.com/protocol-buffers/docs/style
+ *   - Use int64 for job/stage IDs, in case of future extension in Spark core.
+ *   - Use `weakIntern` on string values in create new objects during deserialization.
```

Comment:
`- Use `weakIntern` on string values in create new objects during deserialization.`

This is very important. I think we need a refactor work to fix the current codes

Reply:
cc @panbingkun @techaddict

Reply:
https://issues.apache.org/jira/browse/SPARK-41759 @LuciferYang @gengliangwang